### ISSUE TITLE
Multitask Models fail to generalize properly.

### ISSUE THREAD

**Issue by rbharath** [deepchem/deepchem#180]
When multitask models are built with `SingletaskToMultitask` or with `tensorflow` (and ostensibly keras), the trained models report low numbers for generalization. Since `SingletaskToMultitask` simply trains single-task models and stores them separately, I'm inclined to think this is an error in `Metric` or in `Evaluator`. Need to write some more tests, and debug.

**Comment by rbharath** [comment]
I thinks this problem has now been mostly solved. #189, #186, #182 debugged a number of issues, and I think that multitask models (with up to ~20 tasks) are stable now. I suspect that there are potentially issues with larger datasets (memory problems), but that's not relevant to this issue. Going to go ahead and close.
